src/Illuminate/Database/Eloquent/Concerns/HasEvents.php

```diff
@@ -254,6 +254,26 @@ protected function filterModelEventResults($result)
         return $result;
     }
 
+    /**
+     * @param  \Illuminate\Events\QueuedClosure|\Closure|string|array  $events
+     * @param  \Illuminate\Events\QueuedClosure|\Closure|string|array|null  $callback
+     * @return void
+     */
+    public static function listen($events, $callback = null)
+    {
+        if (! is_string($events) && ! is_array($events)) {
```

Comment:
hey @istiak-tridip, (unless I misunderstand this specific check) as `$events` does not have a declared type, I would expect an error to be thrown let's say if a data type other than string|array is passed as it won't be callable.

Would this make more sense?
 
```php
if ($events instanceof Closure) {
```

Thanks!

Reply:
Thanks, I have updated the condition.